include/deal.II/grid/grid_tools.h

```diff
@@ -4323,9 +4333,12 @@ namespace GridTools
       (void)compute_ghost_owners;
       Assert(false, ExcNeedsMPI());
 #  else
-      constexpr int dim      = MeshType::dimension;
-      constexpr int spacedim = MeshType::space_dimension;
-      auto          tria =
+      constexpr int    dim               = MeshType::dimension;
+      constexpr int    spacedim          = MeshType::space_dimension;
+#    ifdef _MSC_VER
+      static const int spacedim_template = MeshType::space_dimension;
```

Comment:
If constexpr is removed, the MSVC compiler complains about it, and the error occurs: invalid template argument, expected compile-time constant expression.

Reply:
I think what @tjhei is suggesting is that if you write line 4337 as
```
  const int    spacedim          = MeshType::space_dimension;
```
(note: `const` instead of `constexpr`) then that might work. That would be an acceptable change and would not require adding a conditional `#ifdef _MSC_VER` block.

Reply:
@bangerth No, that doesn't work either. You can play around with https://godbolt.org/z/sXDpFo

Reply:
> I think what @tjhei is suggesting is that if you write line 4337 as
> 
> ```
>   const int    spacedim          = MeshType::space_dimension;
> ```
> 
> (note: `const` instead of `constexpr`) then that might work. That would be an acceptable change and would not require adding a conditional `#ifdef _MSC_VER` block.

It should work. But as @marcfehling said, it does not actually due to MSVC compiler's problem